src/shared/components/mutationMapper/MutationMapperStore.ts

```diff
@@ -147,6 +177,66 @@ export default class MutationMapperStore
         }
     }, undefined);
 
+    readonly transcriptsWithAnnotations = remoteData<string[] | undefined>({
+        await: () => [
+            this.indexedVariantAnnotations,
+            this.allTranscripts,
+            this.transcriptsWithProteinLength
+        ],
+        invoke: async()=>{
+            if (this.indexedVariantAnnotations.result && this.allTranscripts.result && this.transcriptsWithProteinLength.result && this.transcriptsWithProteinLength.result.length > 0) {
+                // ignore transcripts without protein length
+                // TODO: better solution is to show only mutations table, not
+                // lollipop plot for those transcripts
+                const transcripts:string[] = _.uniq([].concat.apply([], uniqueGenomicLocations(this.getMutations()).map(
+                    (gl:GenomicLocation) => {
+                        if (this.indexedVariantAnnotations.result && this.indexedVariantAnnotations.result[genomicLocationString(gl)]) {
+                            return this.indexedVariantAnnotations.result[genomicLocationString(gl)].transcript_consequences.map(
+                                (tc:TranscriptConsequence) => tc.transcript_id
+                            ).filter(
+                                (transcriptId: string) => this.transcriptsWithProteinLength.result!!.includes(transcriptId)
+                            );
+                        } else {
+                            return [];
+                        }
+                })));
+                // makes sure the annotations are actually of the form we are
+                // displaying (e.g. nonsynonymous)
+                return transcripts.filter((t:string) => (
+                    getMutationsToTranscriptId(this.getMutations(),
+                                            t,
+                                            this.indexedVariantAnnotations.result!!).length > 0
+                ));
+            } else {
+                return [];
+            }
+        },
+        onError: (err: Error) => {
+            throw new Error("Failed to get transcriptsWithAnnotations");
```

Comment:
Im pretty sure this will only show up in the console. If you want to return this particular error from the mobxpromise, for use in the code, then i think you would have to wrap the `invoke` body with `try..catch` and throw this error manually. Im not 100% on that but I am pretty sure

Reply:
Thanks! Not doing much error handling in current production either atm, console message is ok for now i think. Filed an issue to improve error handling: https://github.com/cBioPortal/cbioportal/issues/5018